src/shared/condition.c

```diff
@@ -1133,8 +1138,8 @@ static int condition_test_psi(Condition *c, char **env) {
         if (r < 0)
                 return log_debug_errno(r, "Failed to parse loadavg: %s", c->parameter);
 
-        r = read_resource_pressure(pressure_path, PRESSURE_TYPE_FULL, &pressure);
-        if (r == -ENODATA) /* cpu.pressure 'full' was added recently, fall back to 'some'. */
+        r = read_resource_pressure(pressure_path, preferred_pressure_type, &pressure);
+        if (r == -ENODATA) /* cpu.pressure 'full' was recently added at cgroup level, fall back to 'some' */
```

Comment:
Please consitionalize this branch on _FULL, since there's no need to fall back if it's already _SOME

Reply:
Sure! For the sake of code clarity or am I missing something? I figured we would never be in such a situation if we've made it this far as 'some' will always return data